src/FakeItEasy/Core/FakeManager.cs

```diff
@@ -93,7 +94,7 @@ public virtual IEnumerable<ICompletedFakeObjectCall> RecordedCallsInScope
             get { return FakeScope.Current.GetCallsWithinScope(this); }
         }
 
-        internal List<ICompletedFakeObjectCall> AllRecordedCalls
+        internal IEnumerable<ICompletedFakeObjectCall> AllRecordedCalls
         {
             get { return this.recordedCallsField; }

```

Comment:
I told you it bordered on nit-pick :stuck_out_tongue: 

I don't feel strongly about this specific case, although I do think that generally, this is where stuff gets messy with properties.

Properties have field semantics, even though they are disguised methods under the hood. I don't expect a property to mutate, but a call to `RecordedCalls()` effectively does that. So, if the object state is mutating, that's fine but the mutation of a property via another method call feels like a side effect. If it is a method that is mutating the internal state, I prefer a method to retrieve that mutated state.

The `ToList()` really is a nitpick. As you say, not so important for internals, but the cost is low (incidentally `ToList()` has been demonstrated as the most efficient way among the other options of `ToArray()`, `Select(_ => _)`, etc. to do this for collection less than thousands of items large). I'd probably insist on the `ToList()` if the method were public.